src/AgentCommerce/SalesChannel/GetCartRoute.php

```diff
@@ -21,9 +25,28 @@
 #[Route(defaults: ['_routeScope' => ['paypal-agent'], '_agentScope' => [AgentSource::SCOPE_CART]])]
 class GetCartRoute
 {
-    #[Route('/api/paypal/v1/merchant-cart/{token}', name: 'api.paypal.merchant-cart.get', methods: [Request::METHOD_GET])]
-    public function getCart(string $token, Request $request, Context $context): AgentCartResponse
+    public function __construct(
+        private readonly CartService $cartService,
+        private readonly PayPalConverter $payPalConverter,
+    ) {
+    }
+
+    #[Route('/api/paypal/v1/merchant-cart/{token}', name: 'api.paypal.merchant-cart.get', defaults: ['auth_required' => false], methods: [Request::METHOD_GET])]
+    public function getCart(string $token, SalesChannelContext $salesChannelContext): AgentCartResponse
     {
-        return new AgentCartResponse($token);
+        if (!\preg_match('/CART-([a-zA-Z0-9]{32})/', $token, $matches)) {
```

Comment:
We should have a central service for this check.
Maybe also use the validator with proper constraint violations.